akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/Effect.scala

```diff
@@ -35,7 +35,10 @@ import akka.util.ccompat.JavaConverters._
    * all events has been persisted. If `callback` is added through [[EffectBuilder.thenRun]] that will invoked
    * after all the events has been persisted.
    */
-  final def persist(events: java.util.List[Event]): EffectBuilder[Event, State] = PersistAll(events.asScala.toVector)
+  final def persist(events: java.util.List[Event]): EffectBuilder[Event, State] = {
+    if (events.size() == 0) none()
+    else PersistAll(events.asScala.toVector)
+  }
```

Comment:
This is a tangent fix, if you persist an empty list, what you are really saying is `Effect.none()`